### ISSUE TITLE
Type relationship query: when the scope function takes parameters

### ISSUE THREAD

**Issue by benjamin-ndc** [rebing/graphql-laravel#561]
### Versions:
- graphql-laravel Version: 4.0.0
- Laravel Version: 5.6.39
- PHP Version: 7.2.4

### Question:

Heya! Enjoying this plugin so far, just have a couple questions about type relationship queries.

First thing, in the docs it says to do this in the type class:

```
class UserType extends GraphQLType {
  ...
  public function fields(): array {
    return [
      ...
      'relationship' => [
        'query' => function(array $args, $query, $ctx) { ... }
      ]
    ]
  }
}
```

However I noticed the function wasn't being called, but if instead of `'query'`, added `'resolve'` like so, it does get called:

```
'resolve' => function($root, $args) { ... }
```

So just wondering if the docs are out of sync or if something else is happening.

**Now for the actual question:** Say you have a scoped relationship that accepts parameters. What I'm noticing is that even if a resolve function is defined, it still complains that the function was called with 0 parameters. So say in my `User` class I have a function like this:

```
public function scopedRelationship($scope) {
   return $this->relationship->where($scope);
}
```

Then in my `UserType` I want to make this scope available, like this:

```
class UserType extends GraphQLType {
  ...
  public function fields(): array {
    return [
      ...
      'scopedRelationship' => [
        'type' => GraphQL::type('relationship'), // already defined elsewhere
        'args' => ['scope' => Type::string()]
        'resolve' => function($root, $args) { return $root->scopedRelationship($args['scope']) }
      ]
    ]
  }
}
```
It bugs out, saying `Too few arguments to function App\\User::scopedRelationship(), 0 passed and at least 1 expected`. So it's like by defining the key `'scopedRelationship'`, it actually calls the User function before calling the resolve function. I'm wondering if this is just by design and I would actually have to define those scopes _at the Type level_, or if there's a way I can leverage the scoped relationships I already have defined?

**Comment by mfn** [comment]
> However I noticed the function wasn't being called, but if instead of `'query'`, added `'resolve'` like so, it does get called

Unfortunately the docs aren't really clear on this distinction:
- `resolve` is the "generic GraphQL way"
- `query` is a special construct in connection only when you use `SelectFields` which is then made use of

>  it actually calls the User function before calling the resolve function

Given your description it certainly looks this way. But exactly because you defined a `resolve` method, this actually should not happen because only the resolver is evaluated.

I think it would help to see the full stacktrace of the exception you get. Or you can also create a repository with a small sample showing the problem.

> Laravel Version: 5.6.39

Btw, glad it works for you it's not an officially supported Laravel version.

**Comment by benjamin-ndc** [comment]
>> Btw, glad it works for you it's not an officially supported Laravel version.

Ahhh, I had more questions but I think I should perhaps start by resolving this one :p thanks for your response!

**Comment by mfn** [comment]
Well, I didn't meant to say that this is the reason it's not supported, but we only test currently active versions (including security updates), see https://laravel.com/docs/6.x/releases#support-policy . 

TL;DR: this only includes 5.5, 5.8 & 6.x
